idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2285 @@
+    "            default=298.15,\n",
+    "            doc=\"Reference temperature\",\n",
+    "            units=units.K,\n",
+    "        )\n",
+    "        self.diffusion_factor = Param(\n",
+    "            self.solute_set,\n",
+    "            initialize={\"NaCl\": 2.15, \"KNO3\": 3, \"CaSO4\": 1.5},\n",
+    "            within=PositiveReals,\n",
+    "            mutable=True,\n",
+    "        )\n",
+    "\n",
+    "    @classmethod\n",
+    "    def define_metadata(cls, obj):\n",
+    "        obj.add_default_units(\n",
+    "            {\n",
+    "                \"time\": units.hour,\n",
+    "                \"length\": units.m,\n",
+    "                \"mass\": units.g,\n",
+    "                \"amount\": units.mol,\n",
+    "                \"temperature\": units.K,\n",
+    "            }\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4c942c2b",
+   "metadata": {},
+   "source": [
+    "# 1.3 State Block\n",
+    "\n",
+    "After the `PhysicalParameterBlock` class has been created, the next step is to write the code necessary to create the State Blocks that will be used throughout the flowsheet. `StateBlock` contains all the information necessary to define the state of the system. This includes the state variables and constraints on those variables which are used to describe a state property like the enthalpy, material balance, etc.\n",
+    "\n",
+    "Creating a State Block requires us to write two classes. The reason we write two classes is because of the inherent nature of how `declare_process_block_data` works. `declare_process_block_data` facilitates creating an `IndexedComponent` object which can handle multiple `ComponentData` objects which represent the component at each point in the indexing set. This makes it easier to build an instance of the model at each indexed point. However, State Blocks are slightly different, as they are always indexed (at least by time). Due to this, we often want to perform actions on all the elements of the indexed StateBlock all at once (rather than element by element).\n",
+    "\n",
+    "The class `_OrganicStateBlock` is defined without the `declare_process_block_data` decorator and thus works as a traditional class and this facilitates performing a method on the class as a whole rather than individual elements of the indexed property blocks. In this class we define the `fix_initialization_states` function. `fix_initialization_states` function is to used to fix the state variable within the state block with the provided initial values (usually inlet conditions). It takes a `block` as the argument in which the state variables are to be fixed. It also takes `state_args` as an optional argument. `state_args` is a dictionary with the value for the state variables to be fixed. This function returns a dictionary indexed by the block, state variables and variable index indicating the fixed status of each variable before applying the function. \n",
```

Comment:
```suggestion
    "The class `_OrganicStateBlock` is defined without the `declare_process_block_data` decorator and thus works as a traditional class and this facilitates performing a method on the class as a whole rather than individual elements of the indexed property blocks. In this class we define the `fix_initialization_states` function. `fix_initialization_states` function is used to fix the state variable within the state block with the provided initial values (usually inlet conditions). It takes a `block` as the argument in which the state variables are to be fixed. It also takes `state_args` as an optional argument. `state_args` is a dictionary with the value for the state variables to be fixed. This function returns a dictionary indexed by the block, state variables and variable index indicating the fixed status of each variable before applying the function. \n",
```